BlueprintUI/Sources/Layout/Decorate.swift

```diff
@@ -0,0 +1,236 @@
+//
+//  Decorate.swift
+//  BlueprintUI
+//
+//  Created by Kyle Van Essen on 11/4/20.
+//
+
+import UIKit
+
+
+///
+/// Places a decoration element behind or in front of the given `wrapped` element,
+/// and positions it according to the `position` parameter.
+///
+/// The size and position of the element is determined only by the `wrapped`
+/// element, the `decoration` element does not affect the layout at all.
+///
+/// Example
+/// -------
+/// The arrows represent the measured size of the element for layout purposes.
+/// ```
+/// ┌───────────────────┐     ┌──────┐
+/// │    Decoration     │     │      │
+/// │ ┏━━━━━━━━━━━━━━━┓ │ ▲   │      ┣━━━━━━━━━━┓   ▲
+/// │ ┃               ┃ │ │   └─┳────┘          ┃   │
+/// │ ┃    Wrapped    ┃ │ │     ┃    Wrapped    ┃   │
+/// │ ┃               ┃ │ │     ┃               ┃   │
+/// │ ┗━━━━━━━━━━━━━━━┛ │ ▼     ┗━━━━━━━━━━━━━━━┛   ▼
+/// └───────────────────┘
+///   ◀───────────────▶         ◀───────────────▶
+/// ```
+public struct Decorate : ProxyElement {
```

Comment:
This looks really close to an implementation of #49, with just a few API tweaks!